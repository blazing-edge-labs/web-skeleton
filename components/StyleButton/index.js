import React, { Component } from 'react'
import PropTypes from 'prop-types'

export default class StyleButton extends Component {
  static propTypes = {
    label: PropTypes.string,
    onToggle: PropTypes.func,
    style: PropTypes.string,
  }

  constructor() {
    super()
    this.onToggle = this.onToggle.bind(this)
  }

  onToggle(e) {
    const { onToggle, style } = this.props

    e.preventDefault()
    onToggle(style)
  }

  render() {
    const { label } = this.props

    return (
      <span onMouseDown={this.onToggle}>{label}</span>
    )
  }
}
