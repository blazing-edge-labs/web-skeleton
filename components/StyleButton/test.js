import React from 'react';
import { shallow } from 'enzyme';
import StyleButton from './';

describe('StyleButton component', () => {
  const onToggle = jest.fn();
  const wrapper = shallow(
    <StyleButton
      label="H1"
      onToggle={onToggle}
      style="header-one"
    />
  );
  const instance = wrapper.instance();

  it('onToggle method', () => {
    const mockPreventDefault = jest.fn();
    const event = {
      preventDefault: mockPreventDefault,
    };
    instance.onToggle(event);

    expect(mockPreventDefault).toHaveBeenCalledTimes(1);
    expect(onToggle).toHaveBeenCalledWith('header-one');
  });
});
