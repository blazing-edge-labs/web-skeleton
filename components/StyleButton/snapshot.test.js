import React from 'react';
import renderer from 'react-test-renderer';
import StyleButton from './';

describe('StyleButton component snapshot', () => {
  it('renders basic required data', () => {
    const tree = renderer.create(
      <StyleButton
        label="H1"
        onToggle={() => {}}
        style="header-one"
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
