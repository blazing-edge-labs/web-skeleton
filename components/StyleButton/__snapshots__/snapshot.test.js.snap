exports[`StyleButton component snapshot renders basic required data 1`] = `
<span
  onMouseDown={[Function]}>
  H1
</span>
`;
